File: Array/9Longest Common Pre/ques9.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string res;

        for (int i=0; i<strs[0].size(); i++){

            char ch= strs[0][i];
            bool flag=true;

            for (int j=1; j<strs.size(); j++){

                if (ch!=strs[j][i]){
                    flag=false;
                    break;
                }


            }
            if (flag==false){
                break;
            }
            else {
                res.push_back(ch);
            }
        }
        return res;
    }
};

File: Array/9Longest Common Pre/ques9.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {

    let res=""
    //traverse first string chars
    for (let i=0; i<strs[0].length; i++){
        //traverse array 
     let ch=strs[0][i]
        let istrue=true;
    
        for (let j=1; j<strs.length; j++){
         if (ch!==strs[j][i]){
             istrue=false
             break;
         }
        }
        if (istrue===false){
            break;
        }
        else
      {  res+=ch;}
    } 
    
    return res;
    };